datajunction-clients/python/datajunction/_internal.py

```diff
@@ -203,13 +203,25 @@ def process_results(results) -> "pd.DataFrame":
     def _get_nodes_in_namespace(
         self,
         namespace: str,
-        type_: Optional[str] = None,
+        type_: Optional[models.NodeType] = None,
     ):
         """
-        Retrieves all nodes in the namespace
+        Retrieves all nodes in given namespace.
         """
         response = self._session.get(
-            f"/namespaces/{namespace}/" + (f"?type_={type_}" if type_ else ""),
+            f"/namespaces/{namespace}/" + (f"?type_={type_.value}" if type_ else ""),
+        )
+        return response.json()
+
+    def _get_all_nodes(
+        self,
+        type_: Optional[models.NodeType] = None,
+    ):
+        """
+        Retrieve all nodes of a given type.
+        """
+        response = self._session.get(
+            "/nodes/" + (f"?node_type={type_.value}" if type_ else ""),
```

Comment:
This is one of two ways to get a list of nodes of a specific type--there's also the /metrics and /dimensions endpoints. I wonder if we should get rid of those so there's just a single way to do this. On a similar note, I wonder if this /nodes endpoint should also have an optional `?namespace=...` argument and then we can get rid of the /namespaces/{namespace}/ endpoint. Not saying this all needs to be done in this PR but curious to hear your thoughts on it.

Reply:
Totally agree with simplifying the server side. We don't need separate listing methods for each node type on the server side. I was actually going to do it in the same PR, just ran out of time. Will do in a follow-up.

Reply:
https://github.com/DataJunction/dj/issues/716